### PR TITLE
Update contributing docs

### PR DESCRIPTION
**Pull request by aliscott**
There's a few things here and probably a lot still to do.

* Removed price explorer scripts in favour of querying MongoDB directly
* Moved docs from price explorer scripts into contributing guide
* Added ToC to contributing guide
* Updated usage-based resource docs to remove the "two levels of support" and update the output examples
* Added docs for Azure creds

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -127,66 +174,119 @@ var ResourceRegistry []*schema.RegistryItem = []*schema.RegistryItem{
...
 
-Finally create a temporary terraform file to test your resource and run (no need to commit that):
+Finally create a temporary Terraform file to test your resource and run (no need to commit that):
 
 ```sh
 make run ARGS="--path my_new_terraform/"
 ```
 
-### Detailed notes
+Please use [this pull request description](https://github.com/infracost/infracost/pull/91) as a guide on the level of details to include in your PR, including required integration tests.
```
Comment (alikhajeh1) [inline CONTRIBUTING.md:183]: FYI I updated https://github.com/infracost/infracost/pull/91 to have `Example output without usage-file:` and `Example output with usage-file:`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -127,66 +174,119 @@ var ResourceRegistry []*schema.RegistryItem = []*schema.RegistryItem{
...
+- Cloud vendor pricing pages for that service (e.g. https://aws.amazon.com/redshift/pricing/)
+- Cloud vendor pricing calculators
+	- AWS: https://calculator.aws
+	- Google: https://cloud.google.com/products/calculator
+	- Azure: https://azure.microsoft.com/en-gb/pricing/calculator/
 
-Please use [this pull request description](https://github.com/infracost/infracost/pull/91) as a guide on the level of details to include in your PR, including required integration tests.
+Our aim is to make Infracost's output understandable without needing to read separate docs. We try to match the cloud vendor pricing webpages as users have probably seen those before. It's unlikely that users will have looked at the pricing service JSON (which comes from cloud vendors' pricing APIs), or looked at the detailed billing CSVs that can show the pricing service names. Please check [this spreadsheet](https://docs.google.com/spreadsheets/d/1H_bn2jLzYr7xyrvNsFn-0rDaGPGpnrVTPsjVHzr-kM4/edit#gid=0) for examples of cost component names and units.
```
Comment (alikhajeh1) [inline CONTRIBUTING.md:195]: I updated the gdoc to fix the cost component names, should we inline that table here? that might remind us to keep it uptodate or add other examples for usage-based resources

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -127,66 +174,119 @@ var ResourceRegistry []*schema.RegistryItem = []*schema.RegistryItem{
...
 
-  ```
-  NAME                                        MONTHLY QTY  UNIT         PRICE   HOURLY COST  MONTHLY COST
+Where a cloud vendor's pricing pages information can be improved for clarify, we'll do that, e.g. on some pricing webpages, AWS mention use "Storage Rate" to describe pricing for "Provisioned IOPS storage", so we use the latter.
```
Comment (alikhajeh1) [inline CONTRIBUTING.md:197]:
```suggestion
Where a cloud vendor's pricing pages information can be improved for clarify, we'll do that, e.g. on some pricing webpages, AWS mention use "Storage Rate" to describe pricing for "Storage (provisioned IOPS SSD)", so we use the latter.
```

**Review by alikhajeh1 — COMMENTED** [review]
Thanks for the iteration :)

**Comment by alikhajeh1** [comment]
LGTM. We should probably add a sentence somewhere about how devs can find PriceHash when writing integration tests.

**Comment by tim775** [comment]
> LGTM. We should probably add a sentence somewhere about how devs can find PriceHash when writing integration tests.

I found the easiest thing was to write the test with an empty pricehash then copy the output from the test failure:
```
 testutil.go:94: 
        	Error Trace:	testutil.go:94
        	Error:      	Not equal: 
        	            	expected: ""
        	            	actual  : "d991ea538483f74fd655b3233c61b427-ee3dd7e4624338037ca6fea0933a662f"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-
        	            	+d991ea538483f74fd655b3233c61b427-ee3dd7e4624338037ca6fea0933a662f
        	Test:       	TestRedshiftCluster
        	Messages:   	unexpected price hash for Backup Storage (first 50 TB)
```

**Comment by aliscott** [comment]
> I found the easiest thing was to write the test with an empty pricehash then copy the output from the test failure

The issue I see with this is that you're relying on the test being correct. Although, this will be less of an issue if we have the hints when writing the integration tests.

**Comment by aliscott** [comment]
Going to merge this for just now to unblock Azure contributors, but the contributing docs in general is something we need to keep iterating on.

**Pull request merged**